src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyExecutionManager.cs

```diff
@@ -210,36 +212,81 @@ private List<ProviderSpecificWorkload<TestRunCriteria>> SplitToWorkloads(TestRun
         // is shared we try to find a workload that is appropriate for it. If we don't find any work that the
         // running testhost can do. Or if the testhost already exited (possibly because of crash), we start another one
         // and give it the next workload.
+
+
+        // List all the sources (dlls) we have and group them by their provider, so we can run either multiple sources
+        // on a single instance of the appropriate testhost (for non-paralell, shared),
+        // or each source in its own instance of the provider (for non-parallel, non-shared, and both parallel shared, and parallel non-shared).
```

Comment:
```suggestion
        // List all the sources (dlls) we have and group them by their provider, so we can run either multiple sources
        // on a single instance of the appropriate testhost (for non-paralell, shared),
        // or each source in its own instance of the provider (for non-parallel, non-shared, and both parallel shared, and parallel non-shared).
        //
        // We prefer running more work in parallel when we can, because doing work on 10 testhosts in
        // parallel is faster than doing the same work on 1 testhost. Even when the sources to discover
        // contain just a few tests.
```

Reply:
added.